### DISCUSSION TITLE
error to run diarization

### DISCUSSION THREAD

**Discussion by Atefeh197** [NVIDIA-NeMo/NeMo#7163] [General]
Hello everyone 
This is my code to do diarization:

`import json
import os
import nemo.collections.asr.models as nemo_asr

from omegaconf import OmegaConf

input_file = '1.wav' # Note : The file needs to be mono .wav
meta = {
            'audio_filepath': input_file,
            'offset': 0,
            'duration':None,
            'label': 'infer',
            'text': '-',
            'num_speakers': 4,
            'rttm_filepath': None, # You can add a reference file here
            'uem_filepath' : None
        }
with open('input_manifest.json','w') as fp:
    json.dump(meta,fp)
    fp.write('\n')
output_dir = os.path.join('my_directory', 'oracle_vad')
os.makedirs(output_dir,exist_ok=True)
MODEL_CONFIG = 'diar_infer_telephonic.yaml'

config = OmegaConf.load(MODEL_CONFIG)
print(OmegaConf.to_yaml(config))
sd_model = nemo_asr(cfg = config)
sd_model.diarize()`

I got this error:
File "D:\PythonCodes\pythonProject\pythonProject\nemo_diar.py", line 28, in <module>
    sd_model = nemo_asr(cfg = config)
TypeError: 'module' object is not callable

anyone knows about that?``

**Comment by tango4j** [comment]
You should not call `nemo.collections.asr.models`.
Please exactly follow the routine we specified in [Speaker Diarization Inference Tutorial
](https://github.com/NVIDIA/NeMo/blob/main/tutorials/speaker_tasks/Speaker_Diarization_Inference.ipynb).
